node/flatpak-node-generator.py

```diff
@@ -900,6 +902,41 @@ async def _handle_node_chromedriver(self, package: Package) -> None:
                                         destination=destination,
                                         only_arches=['x86_64'])
 
+    async def _add_nwjs_cache_downloads(self, version: str, flavor: str = 'normal'):
+        assert not version.startswith('v')
+        nwjs_mirror = 'https://dl.nwjs.io'
+
+        if flavor == 'normal':
+            filename_base = 'nwjs'
+        else:
+            filename_base = f'nwjs-{flavor}'
+
+        destdir = self.gen.data_root / 'nwjs-cache'
+        nwjs_arch_map = [
+            ('x86_64', 'linux-x64', 'linux64'),
+            ('i386', 'linux-ia32', 'linux32'),
+        ]
+        for flatpak_arch, nwjs_arch, platform in nwjs_arch_map:
+            filename = f'{filename_base}-v{version}-{nwjs_arch}.tar.gz'
+            dl_url = f'{nwjs_mirror}/v{version}/{filename}'
+            metadata = await RemoteUrlMetadata.get(dl_url, cachable=True)
+            dest = destdir / f'{version}-{flavor}' / platform
+
+            self.gen.add_archive_source(dl_url,
+                                        metadata.integrity,
+                                        destination=dest,
+                                        only_arches=[flatpak_arch])
+
+    async def _handle_nw_builder(self, package: Package) -> None:
+        if self.nwjs_version:
+            version = self.nwjs_version
+        else:
+            versions_json = json.loads(await Requests.instance.read_all(
+                'https://nwjs.io/versions.json', cachable=False))
```

Comment:
Assuming this would be used with nwbuild's `latest` version, is nwbuild actually able to download the latest without hitting the network every time? If it's not, it would probably be better to either drop this part, or add a note to the README saying to call the builder like `nwbuild -v $(cat blah-blah-blah/nwjs-version)` or similar.

Reply:
Add NW.js note to readme. Hope it's clear enough.